### PR TITLE
add Checks for Super User while viewing Non Members Pages

### PR DESCRIPTION
**Pull request by heyrandhir**
### What is the change?
Issue: https://github.com/Real-Dev-Squad/website-members/issues/367
- [x] Adds a check for Super User while viewing pages of Non-Members. Only Super User can view the Non-Members Page.
- [x] Adds a check for users that do not exist.

### Is it a bug?
 Yes

- Steps to repro
- Expected
- Actual

### \*Dev Tested?

- [x] Yes
- [ ] No

#### Platforms

- [x] Web

#### Browsers

- [x] Chrome
- [ ] Safari
- [ ] Firefox

### Before / After Change Screenshots
Before Anyone can view non-member profile page
image

After:
Non super-users cannot access non-member profile
image

> For visual or interaction changes. Can be video / screenshot.

### CODE REVIEW

**Review by prakashchoudhary07 — APPROVED** [review]
LGTM

**Comment by Maheima** [comment]
LGTM

File: src/pages/[id]/index.js [hunk 1]
```diff
@@ -19,23 +19,35 @@ import {
...
+  const fillActiveTasksArray = async () => {
+    const tasksURL = getActiveTasksURL(id);
+    try {
+      const tasksResponse = await fetch(tasksURL);
+      const { tasks } = await tasksResponse.data;
+      setActiveTasksData(tasks);
+    } catch {
+      setActiveTasksData([]);
+    }
+  };
```
Comment (harshith-venkatesh) [inline src/pages/[id]/index.js:46]: Could you check more on useCallback?
and we could go ahead and add `fillActiveTasksArray` in the dependency there

Comment (harshith-venkatesh) [reply src/pages/[id]/index.js:46]: https://dmitripavlutin.com/dont-overuse-react-usecallback/

I love these blogs along with official docs ofcourse :)

File: src/pages/[id]/index.js [hunk 2]
```diff
@@ -19,23 +19,35 @@ import {
...
-      const { tasks } = await tasksResponse.data;
-      setActiveTasksData(tasks);
-    })();
-  }, [id]);
 
   if (errorMessage) {
     return <NotFound errorMsg={errorMessage} />;
   }
+  if (!user.roles?.member && !isSuperUserMode) {
+    return <NotFound errorMsg="Unauthorized to view this page." />;
```
Comment (harshith-venkatesh) [inline src/pages/[id]/index.js:34]: Is this better to save this as constants if we are planning to utilize it in different pages?

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: src/pages/[id]/index.js (hunk 1 above)
Comment (heyrandhir) [reply src/pages/[id]/index.js]: Thanks for the blog Harshith. It was indeed helpful. I have added the fillActiveTasksArray in the dependency array & wrapped fillActiveTasksArray in useCallBack.

File: src/pages/[id]/index.js (hunk 2 above)
Comment (heyrandhir) [reply src/pages/[id]/index.js]: yes, it's better to add this as a constant. I have created a new constant for the same now :).

**Pull request merged**